test/functional/news_controller_test.rb

```diff
@@ -35,24 +36,49 @@ class NewsControllerTest < ActionController::TestCase
   context "on GET to popular" do
     setup do
       @rubygem4 = create(:rubygem, downloads: 20)
-      create(:version, rubygem: @rubygem4, created_at: 71.days.ago)
+      create(:version, rubygem: @rubygem4, created_at: (Gemcutter::POPULAR_DAYS_LIMIT + 1.day).ago)
       get :popular
     end
 
-    should "not include gems updated since 70 days" do
+    should "not include gems updated prior to Gemcutter::POPULAR_DAYS_LIMIT days ago" do
       refute page.has_content? @rubygem4.name
     end
 
     should "order by gem downloads" do
-      expected_order = [@rubygem3, @rubygem2, @rubygem1]
+      expected_order = [@rubygem3, @rubygem2, @rubygem1].map(&:name)
+      actual_order = assert_select("h2.gems__gem__name").map(&:text)
 
-      assert_select("a.gems__gem > span > h2").each_with_index do |element, index|
-        assert_match(/#{expected_order[index].name}/, element.text)
+      expected_order.each_with_index do |expected_gem_name, i|
+        assert_match(/#{expected_gem_name}/, actual_order[i])
       end
     end
 
     should "display entries and total in page info" do
       assert_select "header > p.gems__meter", text: /.*3 of 100 in total/
     end
   end
+
+  context "with multiple latest versions on different platforms" do
+    setup do
+      Gemcutter::NEWS_PER_PAGE.times do |i|
+        create(:version, rubygem: @rubygem2,
+          created_at: (Gemcutter::NEWS_DAYS_LIMIT - 2.days).ago, platform: "platform#{i}")
+      end
+      get :show
+    end
+
+    should "order by created_at of gem version" do
+      expected_order = [@rubygem2, @rubygem3].map(&:name)
+      actual_order = assert_select("h2.gems__gem__name").map(&:text)
+
+      expected_order.each_with_index do |expected_gem_name, i|
+        assert_match(/#{expected_gem_name}/, actual_order[i])
+      end
+    end
+
+    should "display correct number of entries" do
+      entries = assert_select("h2.gems__gem__name")
+      assert_equal(entries.size, 2)
+    end
+  end
```

Comment:
I included the test for order to account for a potential bug in which duplicate entries are shown for gems with multiple platforms (e.g. for above, that `actual_order` isn't `["Rubygem 2", "Rubygem 2"]`)